src/bootstrap/test.rs

```diff
@@ -994,7 +1001,23 @@ impl Step for Compiletest {
             cmd.arg("--lldb-python-dir").arg(dir);
         }
 
-        cmd.args(&builder.config.cmd.test_args());
+        // Get paths from cmd args
+        let paths = match &builder.config.cmd {
+            Subcommand::Test { ref paths, ..} => &paths[..],
+            _ => &[]
+        };
+
+        // Get test-args by striping suite path
+        let assert_file = |p: &PathBuf| {
+            assert!(p.is_file(), "Expected {:?} to be a path to a test file", p);
```

Comment:
Hi @Mark-Simulacrum Sorry, I missed this. Looks like this panics with `./x.py test suite_path` I think we should just ignore test args if `suite path` is specified. What do you think?